### ISSUE TITLE
Greyhole quit working

### ISSUE THREAD

**Issue by futz** [gboudreau/Greyhole#265]
Greyhole was working. I shut down the NAS for a few days. When I restarted it Greyhole wouldn't work. Actually, samba wasn't working either. I fooled around with things, upgraded server from 18.04 to 20.04. Added 'client min protocol = NT1' and 'client min protocol = NT1' to smb.conf. Now I see my greyhole shares in Network, but can't login.

![trouble](https://user-images.githubusercontent.com/10063058/113517303-f28b0580-9533-11eb-8710-733afc5da3a0.jpg)

**Comment by gboudreau** [comment]
I you can't login on your Greyhole shares, but can connect to non-Greyhole shares, look in the the greyhole.log just after restarting the daemon, and into the Samba logs.

Also, run this after restarting the daemon : https://github.com/gboudreau/Greyhole/wiki/GetHelp#gather-system-information

**Comment by gboudreau** [comment]
Of note : you seem to be running Greyhole with PHP 8, which was not tested. The error you posted above it related to that. Still, that won't prevent you from connecting to Greyhole shares.

**Comment by futz** [comment]
Yup, I can mount non-greyhole shares, but not greyhole shares. Here is what the log said after restarting greyhole:

Apr 04 19:59:19 INFO daemon: Greyhole (version 0.15.5) daemon started.
Apr 04 19:59:19 INFO daemon: Checking MySQL tables...
Apr 04 19:59:19 DEBUG daemon: Loading metadata store backup directories...
Apr 04 19:59:19 DEBUG daemon:   Found 2 directories in the settings table.
Apr 04 19:59:19 DEBUG daemon: Checking symlink at /usr/lib/x86_64-linux-gnu/sam>
Apr 04 19:59:19 DEBUG daemon:   Is OK.
Apr 04 19:59:19 WARN daemon:   Greyhole VFS module (/usr/lib/x86_64-linux-gnu/s>
Apr 04 19:59:19 DEBUG read_smb_spool: Processing Samba spool...
Apr 04 19:59:19 ERROR read_smb_spool: PHP Fatal Error: Uncaught ArgumentCountEr>
Stack trace:
#0 /usr/bin/greyhole(4512): gh_error_handler()
#1 /usr/bin/greyhole(8392): SambaSpool::parse_samba_spool()
#2 /usr/bin/greyhole(8367): DaemonRunner->initialize()
#3 /usr/bin/greyhole(8417): DaemonRunner->run()
#4 {main}
  thrown; BT: greyhole[L5938]

And here is the URL from the information gatherer thing: http://ix.io/2V2g

**Comment by gboudreau** [comment]
> Apr 04 20:03:31 WARN daemon:   Greyhole VFS module (/usr/lib/x86_64-linux-gnu/samba/vfs/greyhole.so) seems to be missing some required libraries. If you have issues connecting to your Greyhole-enabled shares, try to compile a new VFS module for Samba by running this command: /usr/share/greyhole/build_vfs.sh current

Do that (command at the end of the log).

**Comment by futz** [comment]
Had a problem. Looking through the script right now.

futz@junker:/$ sudo /usr/share/greyhole/build_vfs.sh current
[sudo] password for futz: 
Installing build dependencies ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libgnutls28-dev : Depends: libidn2-dev but it is not going to be installed
E: Unable to correct problems, you have held broken packages.

Compiling Greyhole VFS module for samba-4.11.6... 
  Downloading Greyhole source code... 
  Downloading Samba source code... 
/usr/share/greyhole/build_vfs.sh: line 95: cd: samba-4.11.6: No such file or directory

**Comment by gboudreau** [comment]
Looks like the script fails to install dependencies needed to compile the VFS module.
Command it uses is :

```bash
sudo apt-get -y install build-essential python3-dev libgnutls28-dev pkg-config
```
